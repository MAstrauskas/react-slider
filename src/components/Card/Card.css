.card {
  margin: 26px 26px;
  width: 308px;
  height: 323px;
  border: 1px solid rgba(0, 0, 0, 0.08);
  border-radius: 2%;
  -webkit-box-shadow: 10px 10px 81px -35px rgba(0, 0, 0, 0.1);
  -moz-box-shadow: 10px 10px 81px -35px rgba(0, 0, 0, 0.1);
  box-shadow: 10px 10px 81px -35px rgba(0, 0, 0, 0.1);
  background-color: #fff;
}

.card a {
  text-decoration: none;
}

.card-image {
  margin-bottom: 10px;
}
.card-image img {
  object-fit: cover;
  width: 100%;
  height: 133px;
}

.card-header,
.card-text {
  padding-left: 19px;
  padding-right: 19px;
}

/* Puts Icon and Card Title text into one line */
.card-header {
  display: flex;
  flex-flow: row;
}

.card-header-icon {
  margin-top: -2%;
  width: 20%;
  height: 20%;
}

.card-header-title {
  display: flex;
  flex-flow: column;
  margin: 0 3%;
  font-size: 23px;
  font-weight: bold;
  color: #3a3a3a;
}

.card-header-subtitle {
  font-size: 12px;
  font-weight: bold;
  text-transform: uppercase;
  color: #adadad;
}

.card-text p {
  font-size: 12px;
  line-height: 1.583;
  color: #3a3a3a;
}

/* Responsiveness */

/* iPhone 5 and similar resolution phones */
@media (max-width: 330px) {
  .card {
    margin-left: 5px;
    margin-right: 45px;
  }
}
