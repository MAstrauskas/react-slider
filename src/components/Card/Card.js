import React, { Component } from "react";
import ReactHtmlParser from "react-html-parser";
import Like from "../Like/Like";

import Icon from "../../assets/icons/icon.png";

import "./Card.css";

class Card extends Component {
  render() {
    return (
      <div className="card">
        <a href={ReactHtmlParser(this.props.href)}>
          <div className="card-image">
            <img src={ReactHtmlParser(this.props.image)} alt="Random" />
          </div>
          <div className="card-header">
            <img className="card-header-icon" src={Icon} alt="Mindera Icon" />
            <p className="card-header-title">
              {ReactHtmlParser(this.props.title)}
              <span className="card-header-subtitle">
                {ReactHtmlParser(this.props.subtitle)}
              </span>
            </p>
          </div>
          <div className="card-text">
            <p>{ReactHtmlParser(this.props.text)}</p>
          </div>
        </a>
        <Like is_liked={this.props.is_liked} id={this.props.id} />
      </div>
    );
  }
}

export default Card;
