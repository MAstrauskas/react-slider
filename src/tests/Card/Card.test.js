import React from "react";
import { create } from "react-test-renderer";
import Card from "../../components/Card/Card";

describe("Card component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Card/>);
        expect(component.toJSON()).toMatchSnapshot();
    })
});
