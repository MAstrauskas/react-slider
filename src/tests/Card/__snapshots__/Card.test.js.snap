// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component it matches the snapshot 1`] = `
<div
  className="card"
>
  <a>
    <div
      className="card-image"
    >
      <img
        alt="Random"
      />
    </div>
    <div
      className="card-header"
    >
      <img
        alt="Mindera Icon"
        className="card-header-icon"
        src="mindera-icon.png"
      />
      <p
        className="card-header-title"
      >
        <span
          className="card-header-subtitle"
        />
      </p>
    </div>
    <div
      className="card-text"
    >
      <p />
    </div>
  </a>
  <div
    className="like-heart"
    onClick={[Function]}
  >
    <img
      alt="Green Heart Button"
      src="like.svg"
    />
  </div>
</div>
`;
